File: src/main/java/me/mrkirby153/KirBot/command/Command.java
package me.mrkirby153.KirBot.command;

import me.mrkirby153.KirBot.user.Clearance;
import net.dv8tion.jda.core.Permission;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface Command {

    String[] aliases() default {};

    Clearance clearance() default Clearance.USER;

    boolean deleteCallingMessage() default true;

    String description() default "No descrption provided";

    String name();

    Permission[] requiredPermissions() default {Permission.MESSAGE_WRITE};

}

File: src/main/kotlin/me/mrkirby153/KirBot/command/Command.kt
package me.mrkirby153.KirBot.command

import me.mrkirby153.KirBot.user.CLEARANCE_DEFAULT
import net.dv8tion.jda.core.Permission

@Retention(AnnotationRetention.RUNTIME)
annotation class Command(val name: String, val arguments: Array<String> = [],
                         val clearance: Int = CLEARANCE_DEFAULT, val control: Boolean = false,
                         val permissions: Array<Permission> = [])